tests/checks/git.fish

```diff
@@ -80,3 +80,18 @@ set -g __fish_git_prompt_status_order untrackedfiles
 fish_git_prompt
 echo
 #CHECK: (newbranch %)
+
+# Turn on everything and verify we correctly ignore sus config files.
+set -g __fish_git_prompt_status_order stagedstate invalidstate dirtystate untrackedfiles stashstate
+set -g __fish_git_prompt_showdirtystate 1
+set -g __fish_git_prompt_show_informative_status 1
+set -g __fish_git_prompt_showuntrackedfiles 1
+rm -Rf .git *
+git init >/dev/null 2>&1
+echo -n > ran.txt
+git config core.fsmonitor 'echo fsmonitor >> ran.txt; false'
```

Comment:
LGTM, thanks!

(fish_indent would make *some* of the redirections consistent (>/dev/null vs > /dev/null) but it can't see inside quotes)

> Ok, you've convinced me to remove the knob. Our story can be "don't use `bash.showInformativeStatus` on watchman-backed repos" and if someone really needs this they can hack up `fish_git_prompt` themselves.

Yeah. Going back and forth on this one. `bash.showInformativeStatus` and fsmonitor are beyond my horizon so I find it hard to empathize.

> I think it's OK to leave tab completions as they are. If you are tab-completing `git` you probably have a reason, and tab-completing e.g. `git add` will run `git status` which really can benefit from watchman.

OK